src/core_modules/capture-core/components/Pages/MainPage/EventWorkingLists/ColumnSetup/eventWorkingListsColumnSetup.types.js

```diff
@@ -1,16 +1,33 @@
 // @flow
 import type { EventProgram } from '../../../../../metaData';
+import type { EventWorkingListsReduxOutputProps } from '../Redux';
+import type { ColumnConfigs, LoadView, UpdateList } from '../../WorkingLists';
+import type { CustomColumnOrder } from '../types';
 
-export type PassOnProps = $ReadOnly<{
-    listId: string,
-    eventsDataElementValues: Object,
-    eventsMainProperties: Object,
-}>;
-
-export type Props = $ReadOnly<{
-    ...PassOnProps,
+type ExtractedProps = {|
     program: EventProgram,
-    customColumnOrder?: Array<{ id: string, visible: boolean }>,
+    customColumnOrder?: CustomColumnOrder,
     onLoadView: Function,
     onUpdateList: Function,
-}>;
+|};
+
+// had to add customColumnOrder as a non optional type or else it would not be removed. Also, if customColumnOrder is
+// added as non optional to the ExtractedProps only (and not to EventWorkingListsReduxOutputProps),
+// flow complaints about one them being optional.
+type RestProps =$Rest<EventWorkingListsReduxOutputProps & { customColumnOrder: CustomColumnOrder },
+    ExtractedProps & { customColumnOrder: CustomColumnOrder }>;
```

Comment:
Perhaps we can have a call and discuss it? I think will be easier to help me understand than starting a chat here.